src/app/campaign/form/campaign-form.component.ts

```diff
@@ -102,18 +102,24 @@ export class CampaignFormComponent implements OnInit {
     });
   }
 
-  updateCampaignForm({ name, type, status, repName, notes, set_account_uri, set_advertiser_uri }: Campaign) {
-    const findAdvertiserByURI = set_advertiser_uri && this.advertisers &&
-      this.advertisers.find(adv => adv.set_advertiser_uri === set_advertiser_uri);
-    this.campaignForm.patchValue({
-      ...(name && {name}),
-      ...(type && {type}),
-      ...(status && {status}),
-      ...(repName && {repName}),
-      ...(notes && {notes}),
-      ...(set_account_uri && {set_account_uri}),
-      ...(findAdvertiserByURI && {set_advertiser_uri: findAdvertiserByURI.set_advertiser_uri})
-    }, {emitEvent: false, onlySelf: true});
+  updateCampaignForm(campaign: Campaign) {
+    const { name, type, status, repName, notes, set_account_uri, set_advertiser_uri } = campaign;
+    const findAdvertiserByURI =
+      set_advertiser_uri && this.advertisers && this.advertisers.find(adv => adv.set_advertiser_uri === set_advertiser_uri);
+    // only set fields that are present, but allow fields to be explicitly set to null or empty string (new campaign)
```

Comment:
changed to allow the campaign form to be reset for a new campaign with null/empty fields